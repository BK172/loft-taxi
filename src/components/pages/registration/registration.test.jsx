import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Registration from './registration';

describe('Registration component test', () => {
  it('Should render Registration correctly', () => {
    const tree = renderer.create(
      <Registration
        onPageChange={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Form submit', () => {
    const onSubmitAction = jest.fn();
    const wrapper = shallow(
        <Registration
          onPageChange={jest.fn()}
        />
    );

    wrapper.find('.form__container').simulate('submit', {
      preventDefault: jest.fn(),
    });

    expect(onSubmitAction).toHaveBeenCalledTimes(0);
  });

  test('Click on submit button', () => {
    const onSubmitButtonClick = jest.fn();
    const wrapper = shallow(
        <Registration
          onPageChange={jest.fn()}
        />
    );

    wrapper.find('.form__submit-btn').simulate('click', {
      preventDefault: jest.fn(),
    });

    expect(onSubmitButtonClick).toHaveBeenCalledTimes(0);
  });

  test('Click on form footer button', () => {
    const onRegisterButtonClick = jest.fn();
    const wrapper = shallow(
        <Registration
          onPageChange={jest.fn()}
        />
    );

    wrapper.find('.form__btn-footer').simulate('click', {
      preventDefault: jest.fn(),
    });

    expect(onRegisterButtonClick).toHaveBeenCalledTimes(0);
  });
});