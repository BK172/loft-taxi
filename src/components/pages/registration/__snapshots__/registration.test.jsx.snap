// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component test Should render Registration correctly 1`] = `
<div
  className="login-layout"
>
  <div
    className="login-layout__left"
  >
    <img
      alt="LoftTaxi logo"
      src="logo-layout-bg.svg"
    />
  </div>
  <div
    className="login-layout__right"
  >
    <div
      className="MuiPaper-root login-layout__card MuiPaper-elevation5 MuiPaper-rounded"
    >
      <div
        className="form__wrapper"
      >
        <h1
          className="MuiTypography-root MuiTypography-h4"
        >
          Регистрация
        </h1>
        <form
          className="form__container"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Адрес электронной почты
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
            </div>
          </div>
          <div
            className="form__items-two-col"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink={false}
                htmlFor="name"
                id="name-label"
              >
                Имя
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="name"
                  name="firstName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                />
              </div>
            </div>
            <div
              className="form__items-divider"
            />
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink={false}
                htmlFor="surName"
                id="surName-label"
              >
                Фамилия
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="surName"
                  name="surName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Пароль
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
              />
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root form__submit-btn MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Зарегистрироваться
            </span>
          </button>
        </form>
        <p
          className="MuiTypography-root form__btns-container MuiTypography-body1"
        >
          <span>
            Уже зарегистрирован? 
          </span>
          <span
            className="form__items-divider"
          />
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Войти
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
